contracts/modules/LoanMaintenance.sol

```diff
@@ -90,8 +90,15 @@ contract LoanMaintenance is LoanOpeningsEvents, LoanMaintenanceEvents, VaultCont
                 msg.value
             );
         }
+
+        (, uint256 collateralToLoanRate) = IPriceFeeds(priceFeeds).getCurrentMargin(
+            loanParamsLocal.loanToken,
+            loanParamsLocal.collateralToken,
+            loanLocal.principal,
+            loanLocal.collateral
+        );
         
```

Comment:
we don't need to compute the margin here. why don't you just call ```queryRate```? Would be cheaper